utils/wasm-proc/src/main.rs

```diff
@@ -57,10 +55,61 @@ fn main() {
             "meta_init_output",
             "meta_title",
         ],
-    )
-    .expect("Metadata optimizer failed");
-    parity_wasm::serialize_to_file(metadata_file_name.clone(), metadata_module)
-        .expect("Serialization failed");
+    ) {
+        println!("Optimizer failed");
+    }
+
+    if let Err(e) = parity_wasm::serialize_to_file(metadata_file_name.clone(), metadata_module) {
+        println!("Serialization failed: {}", e);
+    }
 
     println!("Metadata wasm: {}", metadata_file_name.to_string_lossy());
 }
+
+fn main() {
+    let meta = Arg::with_name("meta")
+        .short("m")
+        .long("meta")
+        .takes_value(false)
+        .help("Provides a metadata .meta.wasm file");
+
+    let opt = Arg::with_name("optimize")
+        .short("o")
+        .long("optimize")
+        .takes_value(false)
+        .help("Provides an optimized .opt.wasm file");
+
+    let path = Arg::with_name("path")
+        .short("p")
+        .long("path")
+        .required(true)
+        .index(1)
+        .takes_value(true)
+        .multiple(true)
+        .help("Specifies path to .wasm file(-s)");
+
+    let app = App::new("wasm-proc").args(&[meta, opt, path]);
+
+    let matches = app.get_matches();
+
+    let wasm_files: Vec<&str> = matches
+        .values_of("path")
+        .expect("Path to wasm files is required")
+        .collect();
+
+    let o = matches.is_present("optimize");
```

Comment:
```suggestion
    let do_optimize = matches.is_present("optimize");
```

Reply:
Changed according to new (skipping) flags logic